### ISSUE TITLE
IllegalStateException: [GET http://www.0crat.com/?PsByFlag=PsGithub&code=1257b773ba07221e7eb5] failed in 98ms: [400] No ...

### ISSUE THREAD

**Issue by yegor256** [zerocracy/farm#512]
https://sentry.io/zerocracy/farm/issues/438900698/

```
HttpException: [400] No access token
    at org.takes.facets.auth.social.PsGithub.token(PsGithub.java:186)
    at org.takes.facets.auth.social.PsGithub.enter(PsGithub.java:132)
    at org.takes.facets.auth.PsByFlag.enter(PsByFlag.java:107)
    at org.takes.facets.auth.PsChain.enter(PsChain.java:70)
    at org.takes.facets.auth.TkAuth.act(TkAuth.java:86)
...
(25 additional frame(s) were not displayed)

IllegalStateException: [GET http://www.0crat.com/?PsByFlag=PsGithub&code=1257b773ba07221e7eb5] failed in 98ms: [400] No access token
    at com.zerocracy.tk.TkSslOnly.act(TkSslOnly.java:59)
...
(11 additional frame(s) were not displayed)

[GET http://www.0crat.com/?PsByFlag=PsGithub&code=1257b773ba07221e7eb5] failed in 98ms: [400] No access token
```

**Comment by 0crat** [comment]
Job [#512](https://github.com/zerocracy/farm/issues/512) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://datum.zerocracy.com/pages/policy.html#29): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+9130](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#512](https://github.com/zerocracy/farm/issues/512) assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/580913497). The budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4). Please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by g4s8** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/512#issuecomment-370949423))

@g4s8 You can't do that, unless you have one of these roles: `ARC`, `PO`. Your current roles are: `DEV`, `REV`.

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/512#issuecomment-370949423))

@g4s8 The user @g4s8[/z](https://www.0crat.com/u/g4s8) resigned from [#512](https://github.com/zerocracy/farm/issues/512), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 points just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
The job [#512](https://github.com/zerocracy/farm/issues/512) assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/660854893). The budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4). Please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by amihaiemil** [comment]
@yegor256 Can I somehow get access to the error reports on sentry?

**Comment by yegor256** [comment]
@amihaiemil I think you can... sent you an invite

**Comment by amihaiemil** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/512#issuecomment-372578136))

@amihaiemil The user @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) resigned from [#512](https://github.com/zerocracy/farm/issues/512), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
The job [#512](https://github.com/zerocracy/farm/issues/512) assigned to @neonailol[/z](https://www.0crat.com/u/neonailol), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/810872445); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html)

**Comment by neonailol** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/512#issuecomment-376401965))

@neonailol The user @neonailol[/z](https://www.0crat.com/u/neonailol) resigned from [#512](https://github.com/zerocracy/farm/issues/512), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @neonailol[/z](https://www.0crat.com/u/neonailol)

**Comment by carlosmiranda** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/512#issuecomment-411264165))

@carlosmiranda The user @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) resigned from [#512](https://github.com/zerocracy/farm/issues/512), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
The job [#512](https://github.com/zerocracy/farm/issues/512) assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2210754514); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by paulodamaso** [comment]
@yegor256 I need access to error reports regarding zerocracy on sentry, please give some credentials, I've already have a sentry user: `paulo-damaso`

**Comment by paulodamaso** [comment]
@yegor256 @g4s8 I still don't have access to the error reports. I need them so I can see the entire stacktrace.

**Comment by g4s8** [comment]
@paulodamaso only @yegor256 can give you access to sentry, but I can post full stacktrace here:
```
org.takes.HttpException: [400] code is not provided by Github
    at org.takes.facets.auth.social.PsGithub.enter(PsGithub.java:125)
    at org.takes.facets.auth.PsByFlag.enter(PsByFlag.java:107)
    at org.takes.facets.auth.PsChain.enter(PsChain.java:70)
    at org.takes.facets.auth.TkAuth.act(TkAuth.java:86)
    at org.takes.tk.TkWrap.act(TkWrap.java:61)
    at org.takes.facets.flash.TkFlash.act(TkFlash.java:108)
    at org.takes.tk.TkGzip$1.act(TkGzip.java:58)
    at org.takes.tk.TkWrap.act(TkWrap.java:61)
    at org.takes.tk.TkMeasured$1.act(TkMeasured.java:67)
    at org.takes.tk.TkWrap.act(TkWrap.java:61)
    at org.takes.tk.TkVersioned$1.act(TkVersioned.java:73)
    at org.takes.tk.TkWrap.act(TkWrap.java:61)
    at org.takes.tk.TkWithHeaders$1.act(TkWithHeaders.java:71)
    at org.takes.tk.TkWrap.act(TkWrap.java:61)
    at org.takes.facets.fallback.TkFallback.route(TkFallback.java:86)
    at org.takes.facets.fallback.TkFallback.access$000(TkFallback.java:54)
    at org.takes.facets.fallback.TkFallback$1.act(TkFallback.java:66)
    at org.takes.tk.TkWrap.act(TkWrap.java:61)
    at org.takes.tk.TkSslOnly.act(TkSslOnly.java:71)
    at org.takes.tk.TkWrap.act(TkWrap.java:61)
    at org.takes.http.BkBasic.print(BkBasic.java:108)
    at org.takes.http.BkBasic.accept(BkBasic.java:84)
    at org.takes.http.BkSafe$1.accept(BkSafe.java:51)
    at org.takes.http.BkWrap.accept(BkWrap.java:52)
    at org.takes.http.BkTimeable.accept(BkTimeable.java:89)
    at org.takes.http.BkParallel$1$1.run(BkParallel.java:89)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)
java.lang.IllegalStateException: [GET https://www.0crat.com/?PsByFlag=PsGithub] failed in 0ms: [400] code is not provided by Github
    at org.takes.facets.fallback.TkFallback.error(TkFallback.java:234)
    at org.takes.facets.fallback.TkFallback.route(TkFallback.java:92)
    at org.takes.facets.fallback.TkFallback.access$000(TkFallback.java:54)
    at org.takes.facets.fallback.TkFallback$1.act(TkFallback.java:66)
    at org.takes.tk.TkWrap.act(TkWrap.java:61)
    at org.takes.tk.TkSslOnly.act(TkSslOnly.java:71)
    at org.takes.tk.TkWrap.act(TkWrap.java:61)
    at org.takes.http.BkBasic.print(BkBasic.java:108)
    at org.takes.http.BkBasic.accept(BkBasic.java:84)
    at org.takes.http.BkSafe$1.accept(BkSafe.java:51)
    at org.takes.http.BkWrap.accept(BkWrap.java:52)
    at org.takes.http.BkTimeable.accept(BkTimeable.java:89)
    at org.takes.http.BkParallel$1$1.run(BkParallel.java:89)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)
```

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by paulodamaso** [comment]
@0crat wait for #1602 review

**Comment by 0crat** [comment]
> @0crat wait for #1602 review ([here](https://github.com/zerocracy/farm/issues/512#issuecomment-413669894))

@paulodamaso The impediment for [#512](https://github.com/zerocracy/farm/issues/512) was registered successfully by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by paulodamaso** [comment]
@yegor256 Merged, please close.

**Comment by paulodamaso** [comment]
@yegor256 please take a look at this

**Comment by yegor256** [comment]
@paulodamaso thanks!

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#512](https://github.com/zerocracy/farm/issues/512) is now out of scope

**Comment by ypshenychka** [comment]
@paulodamaso According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>All commit messages have links to issues

[One](https://github.com/zerocracy/farm/pull/1602/commits/f0b775fc2e47dc8e66850138ecde81e76933958b) of your commits contains link to another issue. Please confirm you'll try to avoid this mistake in future.

**Comment by paulodamaso** [comment]
@ypshenychka sorry, made some confusion working in parallel with another issue. I confirm that I'll avoid these mistakes in the future.

**Comment by paulodamaso** [comment]
@ypshenychka ping

**Comment by ypshenychka** [comment]
@paulodamaso thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +30 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#unknown]() is still not solved.
